src/slider/__tests__/slider.test.tsx

```diff
@@ -444,4 +444,26 @@ describe('Slider a11y', () => {
 
     expect(labels).toHaveAttribute('aria-hidden', 'true');
   });
+
+  test('aria-readonly is set to true when readOnly', () => {
+    const wrapper = renderSlider({ min: 0, max: 100, value: 50, readOnly: true });
+    const input = wrapper.findNativeInput()?.getElement();
+
+    expect(input).toHaveAttribute('aria-readonly', 'true');
+  });
+
+  test('does not trigger onChange when readOnly', () => {
+    const onChange = jest.fn();
+    const wrapper = renderSlider({
+      min: 0,
+      max: 100,
+      value: 50,
+      onChange: event => onChange(event.detail),
+      readOnly: true,
+    });
+
+    wrapper.setInputValue('60');
```

Comment:
This test is a bit difficult for me. By [definition](https://github.com/cloudscape-design/components/blob/c54515793103c374eb72cf76a52a618e262c960a/src/test-utils/dom/input/base-input.ts#L37-L47) this test util calls `onChange`.

So in theory the test tests what it needs to but it feels like its being tested in an unnatural way. I don't have any better suggestion for you though so we can leave it as is.